### PR TITLE
Fix unfiltered params error when generating ActivityPub tag pagination

### PR DESCRIPTION
**Pull request by Gargron**
`ActionController::UnfilteredParameters (unable to convert unpermitted parameters to hash)`

### CODE REVIEW

File: app/controllers/tags_controller.rb [hunk 1]
```diff
@@ -57,10 +57,14 @@ def set_instance_presenter
...
-      id: tag_url(@tag, params.slice(:any, :all, :none)),
+      id: tag_url(@tag, filter_params),
       type: :ordered,
       size: @tag.statuses.count,
       items: @statuses.map { |s| ActivityPub::TagManager.instance.uri_for(s) }
     )
   end
+
+  def filter_params
+    params.slice(:any, :all, :none).permit(:any, :all, :none)
```
Comment (ThibG) [inline app/controllers/tags_controller.rb]: Why use both `slice` and `permit`? Isn't `permit` enough?

Comment (Gargron) [reply app/controllers/tags_controller.rb]: Permit isn't enough when you want to convert to hash and the hash contains params that are not permitted.

Comment (ThibG) [reply app/controllers/tags_controller.rb]: As far as I understand it, both `slice` and `permit` return an `ActionController::Parameters`, sliced to the params, with `permit` marking them as permitted while `slice` doesn't. So `slice` alone isn't good for that use case, but `permit` is?

Comment (Gargron) [reply app/controllers/tags_controller.rb]:
```
[1] pry(main)> p = ActionController::Parameters.new(foo: '1', bar: '2')
=> <ActionController::Parameters {"foo"=>"1", "bar"=>"2"} permitted: false>
[2] pry(main)> p.permit(:foo).to_h
Unpermitted parameter: :bar
=> {"foo"=>"1"}```

Comment (ThibG) [reply app/controllers/tags_controller.rb]: for some reason i can't reproduce this here, but ok

**Review by ClearlyClaire — APPROVED** [review]

**Pull request merged**